sdk/core/src/shared/interfaces.ts

```diff
@@ -6,4 +6,5 @@ export interface IPlatformEncryptionTools {
   platformName: string
   decryptByPrivateKey(privateKeyBuffer: Buffer, encryptedDataString: string): Promise<any>
   encryptByPublicKey(publicKeyBuffer: Buffer, data: unknown): Promise<string>
+  signWithPrivateKey(privateKeyBuffer: Buffer, data: string): Promise<string>
```

Comment:
What do we need this method for?

If it's needed to transform credential IDs and types for storage, then maybe it would be more appropriate to name it something like `computePersonalHash`.

The way it is named now, it is not clear if two subsequent calls to `signWithPrivateKey` will result in the same value.